### PR TITLE
Refactor lib/Newsletter/ViewInBrowser to Doctrine [MAILPOET-3637]

### PR DESCRIPTION
**Pull request by brezocordero**
Use `NewsletterEntity` and `NewslettersRepository` instead of `MailPoet\Models\Newsletter`.

To test: Edit a Newsletter and preview in browser. Things should work as before

2 integration tests have been also updated to use the Entities instead of the old models.

Run them on the Docker env with:
`./do --test "test:integration --file=tests/integration/Newsletter/ViewInBrowser/ViewInBrowserControllerTest.php"
`
`./do --test "test:integration --file=tests/integration/Newsletter/ViewInBrowser/ViewInBrowserRendererTest.php"
`

I have a question about a function that has a comment saying the function could be removed after the refactoring. Doesn't look like it, see comment on the file below.

[MAILPOET-3637]

[MAILPOET-3637]: https://mailpoet.atlassian.net/browse/MAILPOET-3637

### CODE REVIEW

File: lib/Newsletter/ViewInBrowser/ViewInBrowserRenderer.php [hunk 1]
```diff
@@ -98,20 +95,16 @@ public function render(
 
   /** this is here to prepare entities for the shortcodes library, when this whole file uses doctrine, this can be deleted */
```
Comment (brezocordero) [inline lib/Newsletter/ViewInBrowser/ViewInBrowserRenderer.php:96]: This comment says that the function can be deleted but don't we have to still assign the data to the shortcodes?

File: tests/integration/Newsletter/ViewInBrowser/ViewInBrowserRendererTest.php [hunk 2]
```diff
@@ -136,21 +135,20 @@ public function _before() {
...
-    $newsletterLink1->save();
-
-    $newsletterLink2 = NewsletterLink::create();
-    $newsletterLink2->hash = 'i1893';
-    $newsletterLink2->url = 'http://google.com';
-    $newsletterLink2->newsletterId = $this->newsletter->getId();
-    $newsletterLink2->queueId = $this->sendingTask->id;
-    $newsletterLink2->save();
+
+    $newsletterLink1 = (new \MailPoet\Test\DataFactories\NewsletterLink($newsletter))
```
Comment (lysyjan) [inline tests/integration/Newsletter/ViewInBrowser/ViewInBrowserRendererTest.php:143]: You don't need to create the `NewsletterLink` factory twice. You can create it one time and reuse it. The `with` methods rewrite the previous configuration.
When it is possible, I prefer using the use statement for the fully classified name. What is your opinion on this topic?

Comment (lysyjan) [reply tests/integration/Newsletter/ViewInBrowser/ViewInBrowserRendererTest.php:143]: Also, I would remove the blank line before the link creation.

File: lib/Newsletter/ViewInBrowser/ViewInBrowserRenderer.php (hunk 1 above)
Comment (lysyjan) [reply lib/Newsletter/ViewInBrowser/ViewInBrowserRenderer.php:96]: I agree. In my opinion, we cannot delete it. But maybe I overlooked something.

**Review by lysyjan — CHANGES_REQUESTED** [review]
Good job, but I have few notices.

File: tests/integration/Newsletter/ViewInBrowser/ViewInBrowserRendererTest.php (hunk 2 above)
Comment (brezocordero) [reply tests/integration/Newsletter/ViewInBrowser/ViewInBrowserRendererTest.php]: Done!

File: tests/integration/Newsletter/ViewInBrowser/ViewInBrowserRendererTest.php [hunk 3]
```diff
@@ -108,9 +111,15 @@ public function _before() {
     ];
 
     // create newsletter
-    $newsletter = Newsletter::create();
-    $newsletter->hydrate($newsletterData);
-    $this->newsletter = $newsletter->save();
+    $newsletter = new NewsletterEntity();
```
[the remark this answers is not in the document]
Comment (brezocordero) [reply tests/integration/Newsletter/ViewInBrowser/ViewInBrowserRendererTest.php]: Done!

File: tests/integration/Newsletter/ViewInBrowser/ViewInBrowserControllerTest.php [hunk 4]
```diff
@@ -66,17 +125,17 @@ public function _before() {
 
     // create task & queue
     $sendingTask = SendingTask::create();
-    $sendingTask->newsletterId = $newsletter->id;
+    $sendingTask->newsletterId = $newsletter->getId();
     $sendingTask->setSubscribers([$subscriber->getId()]);
     $sendingTask->updateProcessedSubscribers([$subscriber->getId()]);
     $this->sendingTask = $sendingTask->save();
-
+    $this->newslettersRepository->refresh($newsletter);
```
[the remark this answers is not in the document]
Comment (brezocordero) [reply tests/integration/Newsletter/ViewInBrowser/ViewInBrowserControllerTest.php]: Removed.

File: lib/Newsletter/ViewInBrowser/ViewInBrowserRenderer.php (hunk 1 above)
Comment (brezocordero) [reply lib/Newsletter/ViewInBrowser/ViewInBrowserRenderer.php]: Thanks for confirming, I removed the comment.

**Review by lysyjan — APPROVED** [review]
Good job!

**Review by veljkho — APPROVED** [review]

**Pull request merged**